File: apps/ios/Bedrud/Models/Instance.swift
import Foundation

struct Instance: Codable, Identifiable, Equatable {
    let id: String
    let serverURL: String
    let displayName: String
    let iconColorHex: String
    let addedAt: Date

    init(
        id: String = UUID().uuidString,
        serverURL: String,
        displayName: String,
        iconColorHex: String = Self.randomColor(),
        addedAt: Date = Date()
    ) {
        self.id = id
        self.serverURL = serverURL
        self.displayName = displayName
        self.iconColorHex = iconColorHex
        self.addedAt = addedAt
    }

    /// The API base URL derived from the server URL.
    var apiBaseURL: String {
        serverURL.hasSuffix("/") ? "\(serverURL)api" : "\(serverURL)/api"
    }

    private static func randomColor() -> String {
        let colors = ["#3B82F6", "#EF4444", "#10B981", "#F59E0B", "#8B5CF6", "#EC4899", "#06B6D4", "#F97316"]
        return colors.randomElement()!
    }
}

struct Account: Codable, Equatable {
    let instanceId: String
    var userId: String?
    var userName: String?
    var userEmail: String?
    var isLoggedIn: Bool
}

struct HealthResponse: Decodable {
    let status: String?
    let version: String?
}

File: apps/android/app/src/main/java/com/bedrud/app/models/Instance.kt
package com.bedrud.app.models

data class Instance(
    val id: String = java.util.UUID.randomUUID().toString(),
    val serverURL: String,
    val displayName: String,
    val iconColorHex: String = randomColor(),
    val addedAt: Long = System.currentTimeMillis()
) {
    val apiBaseURL: String
        get() {
            val base = if (serverURL.endsWith("/")) serverURL else "$serverURL/"
            return "${base}api"
        }
}

data class Account(
    val instanceId: String,
    val userId: String? = null,
    val userName: String? = null,
    val userEmail: String? = null,
    val isLoggedIn: Boolean = false
)

data class HealthResponse(
    val status: String? = null,
    val version: String? = null
)

private fun randomColor(): String {
    val colors = listOf("#3B82F6", "#EF4444", "#10B981", "#F59E0B", "#8B5CF6", "#EC4899", "#06B6D4", "#F97316")
    return colors.random()
}
